src/main/kotlin/com/buildit/bookit/v1/location/bookable/BookableController.kt

```diff
@@ -67,14 +72,21 @@ class BookableController(private val bookableRepository: BookableRepository, pri
             end.atStartOfDay(timeZone).toInstant()
         )
 
+        val allBookings = bookingRepository.getAllBookings()
         return bookableRepository.getAllBookables()
             .filter { it.locationId == locationId }
             .map { bookable ->
                 val bookings = when {
                     "bookings" in expand ?: emptyList() ->
-                        bookingRepository.getAllBookings()
+                        allBookings
                             .filter { booking -> booking.bookableId == bookable.id }
                             .filter { desiredInterval.overlaps(it.interval(timeZone)) }
+                            .map { booking ->
+                                if (booking.user.externalId == user.subject)
```

Comment:
is comparing the externalId good enough here so i don't have to go back to the database?  If so, probably could do the same for DELETE /booking

Reply:
Seems good enough to me, and natural in the context.  But why did we bother creating a local user ID and mapping it, in that case?  Not a challenge to you, but more food for thought.

Reply:
well, we need to be able to show who has a bookable booked.  So you either make that part of the booking table or normalize it a bit to a users table... i still think that's the right call... and if you do that, then it makes more sense to have an internal id be the PK/FK...but the "business key" is still the external id... plus it probably helps when we add admin users and/or "limit to only *these* users"

Reply:
OK, local ID more for capturing relationships than necessarily for selection, might be a way to say it?  I'm good with it.  I think ;-)